drivers/serial/uart_stm32.c

```diff
@@ -905,6 +905,10 @@ static int uart_stm32_async_rx_disable(const struct device *dev)
 	data->rx_next_buffer = NULL;
 	data->rx_next_buffer_len = 0;
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
```

Comment:
Since `LL_USART_DisableIT_RXNE` is not done under this switch in `uart_stm32_async_rx_enable`, let's not use here here for coherency sake.
If a IRQ config flag clean up is required, let's do it in a dedicated PR.

Reply:
- there's no switch `uart_stm32_async_rx_enable` defined inside the function. switch you referring to?
- `LL_USART_DisableIT_RXNE` was actually called under `uart_stm32_async_rx_enable` , take a look permalink from intact main code, line 1084
https://github.com/zephyrproject-rtos/zephyr/blob/698a28663cb364b0601eabd48c73a206fe1f715c/drivers/serial/uart_stm32.c#L1061-L1092

sorry @erwango I think I may not understand you, please explain if my understanding is wrong.

Reply:
I think the code addition is correct.
My request is to remove the `#ifdef CONFIG_UART_INTERRUPT_DRIVEN` directive for coherency with the rest of the code.